omod/src/main/java/org/openmrs/module/appui/UiSessionContext.java

```diff
@@ -46,6 +48,33 @@ public UiSessionContext() {
 
     }
 
+    public String getViewProvidersPrivilege()
+    {
+    	PrivilegeConstants privilegeConstants = new PrivilegeConstants();
+    	String viewProvidersPrivilege = "";
+    	try {
+    		// openmrsVersion 1.9
+    		Class<?> privilegeConstantsClass = privilegeConstants.getClass();
+    		Field viewProvidersPrivilegeConstant = privilegeConstantsClass.getDeclaredField("VIEW_PROVIDERS");
+    		viewProvidersPrivilege = (String)viewProvidersPrivilegeConstant.get(privilegeConstants);
+    	} catch (NoSuchFieldException ex) {
+    		// do nothing
+ 		} catch (IllegalAccessException ex) {
+			// do nothing
+		}
+    	try {
+    		// openmrsVersion 2.0
+    		Class<?> privilegeConstantsClass = privilegeConstants.getClass();
+    		Field viewProvidersPrivilegeConstant = privilegeConstantsClass.getDeclaredField("GET_PROVIDERS");
+    		viewProvidersPrivilege = (String)viewProvidersPrivilegeConstant.get(privilegeConstants);
+    	} catch (NoSuchFieldException ex) {
+    		// do nothing
+ 		} catch (IllegalAccessException ex) {
+			// do nothing
+		}
+    	return viewProvidersPrivilege;
+    }
+    
```

Comment:
It feels like a hacky way to do things -- It has been ages since I touched OpenMRS Core, but there has to be a way to read the version and use it as a conditional...it's not a cheap operation...plus exceptions should not be used in this manner...especially when there is a better way...This is just my opinion however.